go/dpagg/count_test.go

```diff
@@ -345,6 +345,93 @@ func TestCountThresholdedResult(t *testing.T) {
 	}
 }
 
+func TestCountComputeConfidenceIntervalPostProcessing(t *testing.T) {
+	// Confidence interval bounds for count should be non-negative.
+	for _, tc := range []struct {
+		confInt noise.ConfidenceInterval // Pre-processing.
+		want    noise.ConfidenceInterval // Post-processing.
+	}{
+		{
+			confInt: noise.ConfidenceInterval{LowerBound: 5, UpperBound: 10},
+			want:    noise.ConfidenceInterval{LowerBound: 5, UpperBound: 10},
+		},
+		{
+			confInt: noise.ConfidenceInterval{LowerBound: -5, UpperBound: 5},
+			want:    noise.ConfidenceInterval{LowerBound: 0, UpperBound: 5},
+		},
+		{
+			confInt: noise.ConfidenceInterval{LowerBound: -5, UpperBound: -5},
+			want:    noise.ConfidenceInterval{LowerBound: 0, UpperBound: 0},
+		},
+		// Infinite bounds happens for extremely small alpha for Gaussian.
+		{
+			confInt: noise.ConfidenceInterval{LowerBound: math.Inf(-1), UpperBound: math.Inf(1)},
+			want:    noise.ConfidenceInterval{LowerBound: 0, UpperBound: math.Inf(1)},
+		},
+	} {
+		c := getNoiselessCount()
+		// This makes Noise interface return the pre-processing confidence interval when ComputeConfidenceIntervalInt64 is called.
+		c.noise = getMockConfInt(tc.confInt)
+
+		c.Result()
+		got, _ := c.ComputeConfidenceInterval(0.1) // alpha is ignored in mockConfInt.
+
+		if got.LowerBound != tc.want.LowerBound {
+			t.Errorf("TestCountComputeConfidenceIntervalPostProcessing(ConfidenceInterval{%f, %f})=%0.10f, want %0.10f, LowerBounds are not equal",
+				tc.confInt.LowerBound, tc.confInt.UpperBound, got.LowerBound, tc.want.LowerBound)
+		}
+		if got.UpperBound != tc.want.UpperBound {
+			t.Errorf("TestCountComputeConfidenceIntervalPostProcessing(ConfidenceInterval{%f, %f})=%0.10f, want %0.10f, UpperBounds are not equal",
+				tc.confInt.LowerBound, tc.confInt.UpperBound, got.UpperBound, tc.want.UpperBound)
+		}
+	}
+}
+
+func TestCountComputeConfidenceIntervalComputation(t *testing.T) {
+	// Tests returned confidence intervals using Laplace or Gaussian for a given count.
```

Comment:
Again, please move this above function signature: `Tests ComputeConfidenceInterval returns a correct interval for a given count.`